### PR TITLE
Refactor infrastructure to use Azure Flexible SQL as server db

### PR DESCRIPTION
**Pull request by tiagojsag**
### Overview

_Please write a description. If the PR is hard to understand, provide a quick explanation of the code._

### Designs

_Link to the related design prototypes (if applicable)_

### Testing instructions

_Please explain how to test the PR: ID of a dataset, steps to reach the feature, etc._

### Feature relevant tickets

_Link to the related task manager tickets_

---

## Checklist before submitting

- [ ] Meaningful commits and code rebased on `develop`.
- [ ] If this PR adds feature that should be tested for regressions when
      deploying to staging/production, please add brief testing instructions
      to the deploy checklist (`docs/deployment-checklist.md`)
- [ ] Update CHANGELOG file

### CODE REVIEW

File: infrastructure/base/modules/database/main.tf [hunk 1]
```diff
@@ -0,0 +1,39 @@
+locals {
+  postgres_secret_json = {
+    username = "postgres"
+    password = random_password.postgresql_admin_generator.result
+  }
```
Comment (hotzevzl) [inline infrastructure/base/modules/database/main.tf:5]: should we rely on the stock `postgres` user?
in practice, this should be equivalent to the current setup where we create an ad-hoc user, except perhaps that `postgres` may have additional permissions granted via `pg_hba` (on the `postgres` db, for example, though in practice I can successfully `\c postgres` as `marxan-api` user already, in a local db), and also we would forego (if that's an issue to start with - probably not) the separation of users and permissions across `apidb` and `geodb` when they are within the same physical PostgreSQL server.

tl;dr I think it's ok, just listing some things to cross check.

**Review by hotzevzl — APPROVED** [review]

File: infrastructure/base/modules/database/main.tf (hunk 1 above)
Comment (hotzevzl) [reply infrastructure/base/modules/database/main.tf:5]: ah right, this is the `base` layer - all good then.

File: infrastructure/base/vars/terraform.tfvars [hunk 2]
```diff
@@ -7,3 +7,10 @@ github_org               = "Vizzuality"
 github_repo              = "marxan-cloud"
 github_production_branch = "main"
 github_staging_branch    = "staging"
+key_vault_access_users  = ["tiago.garcia_vizzuality.com#EXT#", "andrea.rota_vizzuality.com#EXT#"]
+
+production_db_instance_size = "GP_Standard_D32d_v4"
```
Comment (hotzevzl) [inline infrastructure/base/vars/terraform.tfvars:12]: @tiagojsag just noticed this while trying to remember how many cores I could try to max out: shouldn't this be `GP_Standard_D32ds_v4`?
